packages/core/src/config/config.ts

```diff
@@ -566,6 +566,11 @@ export class Config {
   }
 
   setApprovalMode(mode: ApprovalMode): void {
+    if (!this.isTrustedFolder() && mode !== ApprovalMode.DEFAULT) {
```

Comment:
this is called in edit.ts and a few other places. Please verify that all those places work correctly. Likely a few of them need to be fixed by removing the option from the list of radio buttons to "approve always"
if you search for `setApprovalMode` in the project you will find all the missing cases.

Reply:
Thanks, checked that there are 3 other places in core - write_file.ts, edit.ts and web-fetch.ts that all call setApprovalMode with Auto edit in case of PROCEED_ALWAYS option selection.

Updated ToolConfirmationMessage to skip this option for untrusted folders. I also skipped it in other cases like exec tool calls and mcp tool calls, though these paths do not call setApprovalMode. Added a screenshot with the updated options in the description.